### ISSUE TITLE
Bug - Main Calendar endpoint returns Unknown Property error

### ISSUE THREAD

**Issue by c-nv-s** [humhub/humhub-modules-rest#49]
the base calendar endpoint "your.site/api/v1/calendar" returns:
```
{
  "name": "Unknown Property",
  "message": "Getting unknown property: humhub\\modules\\calendar\\models\\CalendarEntry::recur",
  "code": 0,
  "type": "yii\\base\\UnknownPropertyException",
  "file": "/HumHub/protected/vendor/yiisoft/yii2/base/Component.php",
  "line": 154,
  "stack-trace": [
    "#0 /HumHub/protected/vendor/yiisoft/yii2/db/BaseActiveRecord.php(298): yii\\base\\Component->__get('recur')",
    "#1 /HumHub/protected/humhub/modules/content/components/ContentActiveRecord.php(192): yii\\db\\BaseActiveRecord->__get('recur')",
    "#2 /HumHub/protected/modules/rest/definitions/CalendarDefinitions.php(29): humhub\\modules\\content\\components\\ContentActiveRecord->__get('recur')",
    "#3 /HumHub/protected/modules/rest/controllers/calendar/CalendarController.php(39): humhub\\modules\\rest\\definitions\\CalendarDefinitions::getCalendarEntry(Object(humhub\\modules\\calendar\\models\\CalendarEntry))",
    "#4 /HumHub/protected/modules/rest/components/BaseContentController.php(76): humhub\\modules\\rest\\controllers\\calendar\\CalendarController->returnContentDefinition(Object(humhub\\modules\\calendar\\models\\CalendarEntry))",
    "#5 [internal function]: humhub\\modules\\rest\\components\\BaseContentController->actionFind()",
    "#6 /HumHub/protected/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)",
    "#7 /HumHub/protected/vendor/yiisoft/yii2/base/Controller.php(180): yii\\base\\InlineAction->runWithParams(Array)",
    "#8 /HumHub/protected/vendor/yiisoft/yii2/base/Module.php(528): yii\\base\\Controller->runAction('find', Array)",
    "#9 /HumHub/protected/vendor/yiisoft/yii2/web/Application.php(103): yii\\base\\Module->runAction('rest/calendar/c...', Array)",
    "#10 /HumHub/protected/vendor/yiisoft/yii2/base/Application.php(386): yii\\web\\Application->handleRequest(Object(humhub\\components\\Request))",
    "#11 /HumHub/index.php(25): yii\\base\\Application->run()",
    "#12 {main}"
  ]
}
```

expected example result : 
```
{
  "total": 76,
  "page": 1,
  "pages": 8,
  "links": {
    "self": "{endpoint_path}?page=1&per-page=10",
    "next": "{endpoint_path}?page=2&per-page=10",
    "last": "{endpoint_path}?page=8&per-page=10"
  },
  "results": [
    {
      "id": 6,
      "title": "Space event by API",
      "description": "Space event by API description",
      "start_datetime": "2019-02-27 00:00:00",
      "end_datetime": "2019-02-28 23:59:00",
      "all_day": 1,
      "participation_mode": 2,
      "color": "#6fdbe8",
      "allow_decline": 1,
      "allow_maybe": 1,
      "time_zone": "America/Los_Angeles",
      "participant_info": "",
      "closed": 0,
      "max_participants": null,
      "content": {
        "metadata": {
          "id": 14,
          "guid": "cd081891-d2e2-40d5-84a4-b47309e71c80",
          "url": "http://localhost/p/humhub/master/content/perma?id=109",
          "created_by": {
            "id": 14,
            "guid": "cd081891-d2e2-40d5-84a4-b47309e71c80",
            "display_name": "John Doe",
            "url": "http://localhost/p/humhub/master/u/j.doe/"
          },
          "created_at": null,
          "contentcontainer_id": 5,
          "stream_channel": "default"
        },
        "comments": {
          "total": 55,
          "latestComments": [
            {
              "id": 14,
              "message": "Calvin Klein – Between love and madness lies obsession.",
              "createdBy": {
                "id": 14,
                "guid": "cd081891-d2e2-40d5-84a4-b47309e71c80",
                "display_name": "John Doe",
                "url": "http://localhost/p/humhub/master/u/j.doe/"
              },
              "createdAt": "2020-11-23T10:17:07.586Z"
            }
          ]
        },
        "likes": {
          "total": 35
        },
        "files": {
          "id": 14,
          "guid": "bded6dc7-e6f4-417e-a638-b653525529ca",
          "mime_type": "text/plain",
          "size": 14353,
          "file_name": "image.jpg",
          "url": "http://localhost/p/humhub/master/file/file/download?guid=bded6dc7-e6f4-417e-a638-b653525529ca"
        },
        "topics": {
          "topic": [
            null
          ]
        }
      }
    }
  ]
}
```

**Comment by luke-** [comment]
Are you running the latest version of the Calendar module?

**Comment by c-nv-s** [comment]
![humhub-calendarversion](https://user-images.githubusercontent.com/7495833/108052724-d6e3a280-704c-11eb-950e-55666c04bdfd.png)

I will remove and re-install to double check

**Comment by c-nv-s** [comment]
It may be that upgrading the plugin or the system through the "Module Administration" page didn't properly update the plugin. 
I had to re-install the calendar plugin for this to work.
This is something which I am not able to do on the production machine without loosing all the events (unless there is a bulk export/import feature), and is also returning a 500 error which I'm still looking into.

**Comment by luke-** [comment]
So the original problem is fixed? Maybe the database migrations did not run correctly. Is there an error message about this?

**Comment by buddh4** [comment]
@c-nv-s go to  `Administration -> Information -> Database` and check if there are any errors.

**Comment by c-nv-s** [comment]
there are no migration errors reported at all. just in the logging it shows the same error as i first reported everytime you make a call to the base endpoint.

it's a little strange because calling the container endpoint doesn't seem to error
```
$  GET mysite.com/api/v1/calendar/container/1  Header "Authorization:Bearer ${TOKEN}"  

{
  "total": "0",
  "page": 1,
  "pages": 0,
  "links": {
    "self": "/api/v1/calendar/container/1?page=1&per-page=100"
  },
  "results": []
}

```

**Comment by buddh4** [comment]
Did you by any chance manually add a field to https://github.com/humhub/humhub-modules-rest/blob/master/definitions/CalendarDefinitions.php#L29 maybe in order to support recurring events?

**Comment by c-nv-s** [comment]
@buddh4 no I have not touched any of that.

**Comment by buddh4** [comment]
It seems this is already fixed in master and was fixed by: https://github.com/humhub/humhub-modules-rest/commit/5d9e301030d6e6d66c3b29cc47a3bba426c56a43#diff-934e23a8ae2c4555ab7c3977a6e0749f967403a9e99b8c54aee655a3a58290f7

I've opened an issue to add the actual recurrent event fields here: https://github.com/humhub/humhub-modules-rest/issues/50

**Comment by luke-** [comment]
Ah ok, we were still in a beta phase here.
I have now released the stable version 0.2.1 of the Rest module. This should contain the fix.
